src/main/java/com/iridium/iridiumskyblock/listeners/PlayerRespawnEventListener.java

```diff
@@ -0,0 +1,20 @@
+package com.iridium.iridiumskyblock.listeners;
+
+import com.iridium.iridiumskyblock.IridiumSkyblock;
+import com.iridium.iridiumskyblock.utils.PlayerUtils;
+import lombok.AllArgsConstructor;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.Listener;
+import org.bukkit.event.player.PlayerRespawnEvent;
+
+@AllArgsConstructor
+public class PlayerRespawnEventListener implements Listener {
+
+    @EventHandler(priority = EventPriority.HIGHEST)
+    public void onPlayerRespawnEvent(PlayerRespawnEvent event) {
+
+        if(!IridiumSkyblock.getInstance().getConfiguration().spawnOnIsland) { return; }
+        if(!event.isBedSpawn() && !event.isAnchorSpawn()) event.setRespawnLocation(PlayerUtils.getSpawn(event.getPlayer()));
```

Comment:
Is it worth checking if the current spawn is inside the Island?

If we set our bed spawn at another players island, would we expect to respawn in somone elses island or at our island home?

Reply:
🤔 im not 100% sure tbh, because you'd expect that someone would respawn at their bed regardless of where that bed is, since that is vanilla behavior
we can add a check to the interact permission to prevent players from setting their spawn on other player islands if this is a concern

Reply:
I think a check IridiumSkyblock.getInstance().getIslandManager().canVisit should be added